stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1222 @@
+import json
+import requests_mock
+import logging
+from enum import Enum
+from schematics.models import Model
+from schematics.types import BaseType, StringType, IntType, DictType, BooleanType
+from schematics.exceptions import DataError
+from requests import Session, Request
+from requests.auth import HTTPBasicAuth
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+
+class HTTPBasicAuthentication(Model):
+    username = StringType(required=True)
+    password = StringType(required=True)
+
+
+class HTTPAuthenticationType(Enum):
+    BasicAuth = HTTPBasicAuthentication
+
+
+class HTTPMethod(Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+class HTTPRequestType(Enum):
+    JSON = dict
+
+
+class HTTPResponseType(Enum):
+    PLAIN = str
+    JSON = dict
+
+
+class HTTPHelperModel(Model):
+    """
+        This is the model for the `HTTPHelper` class.
+        When using the `HTTPHelper` class there will be the only acceptable values to create state and kick off the
+        request
+    """
+    mock_enable = BooleanType(required=False, default=False)
+    mock_status = IntType(required=False, default=None)
+    mock_response = BaseType(required=False, default=None)
+    endpoint = StringType(required=True, default=None)
+    auth_data = DictType(BaseType, required=False, default=None)
+    auth_type = BaseType(required=False, default=None)
+    session_auth_data = BaseType(required=False, default=None)
+    session_auth_type = BaseType(required=False, default=None)
```

Comment:
I wonder what's the point of having session be part of this model. If you use this model you only get the response back, but you won't actually get the session back, therefore you can't actually use session this way. 

You should consider how a user of this lib is going to use sessions.

Reply:
Added a response Connection response handler which will be the last point of exposure for the user. get_session is added to that